code/controllers/subsystems/trade.dm

```diff
@@ -174,38 +176,69 @@ SUBSYSTEM_DEF(trade)
 
 	station.generate_offer()
 
+/datum/controller/subsystem/trade/proc/collect_counts_from(list/m)
+	. = 0
+	if(islist(m))
+		for(var/n in m)
+			var/list/c = m[n]
+			for(var/t in c)
+				var/tcount = c[t]
+				if(isnum(tcount))
+					. += tcount
+
+/datum/controller/subsystem/trade/proc/collect_price_for_list(list/m)
+	. = 0
+	for(var/t in m)
+		if(ispath(t))
+			. += get_import_cost(t) * get_2d_matrix_cell(m, t, "count")
+
+/datum/controller/subsystem/trade/proc/shoplist2list(list/m)
+	if(islist(m))
+		. = list()
```

Comment:
I know you're being fancy with pointers, but please, just return dot (or hopefully whatever you rename it to).